src/redfishpower/redfishpower.c

```diff
@@ -754,6 +757,112 @@ static void setpowerpath(char **av, char **path, char **postdata)
     }
 }
 
+static void remove_initial_plugs(void)
+{
+    hostlist_iterator_t itr;
+    char *hostname;
+
+    if (!initial_plugs_setup)
+        return;
+
+    if (!(itr = hostlist_iterator_create(hosts)))
+        err_exit(true, "hostlist_iterator_create");
+
+    while ((hostname = hostlist_next(itr))) {
+        plugs_remove(plugs, hostname);
+        free(hostname);
+    }
+
+    hostlist_iterator_destroy(itr);
+    initial_plugs_setup = 0;
+    return;
+}
+
+static void setplugs(char **av)
+{
+    hostlist_t lplugs = NULL;
+    hostlist_t indices = NULL;
+    int plugcount, indexcount;
+    hostlist_iterator_t lplugsitr = NULL;
+    hostlist_iterator_t indicesitr = NULL;
+    char *plug;
+    char *indexstr;
+
+    if (!av[0] || !av[1]) {
+        printf("Usage: setplugs <plugnames> <hostindices>\n");
+        return;
+    }
+
+    if (!(lplugs = hostlist_create(av[0]))) {
+        printf("illegal plugs input\n");
+        goto cleanup;
+    }
+    if (!(indices = hostlist_create(av[1]))) {
+        printf("illegal index input\n");
+        goto cleanup;
+    }
+
+    if ((plugcount = hostlist_count(lplugs)) < 0)
+        err_exit(true, "hostlist_count error");
+    if ((indexcount = hostlist_count(indices)) < 0)
+        err_exit(true, "hostlist_count error");
+    if (plugcount != indexcount) {
+        printf("plugs count not equal to index count");
+        goto cleanup;
+    }
+
+    if (!(lplugsitr = hostlist_iterator_create(lplugs)))
+        err_exit(true, "hostlist_iterator_create");
+    if (!(indicesitr = hostlist_iterator_create(indices)))
+        err_exit(true, "hostlist_iterator_create");
```

Comment:
These functions can't fail.

Note that the behavior of all the libsd functions on ENOMEM is configurable by setting WITH_LSD_NOMEM_ERROR_FUNC which we do project wide.